parser-typechecker/src/Unison/Codebase/Editor/RemoteRepo.hs

```diff
@@ -4,36 +4,51 @@
 module Unison.Codebase.Editor.RemoteRepo where
 
 import qualified Data.Text as Text
-import qualified Servant.Client as Servant
 import qualified U.Util.Monoid as Monoid
 import Unison.Codebase.Path (Path)
 import qualified Unison.Codebase.Path as Path
 import Unison.Codebase.ShortBranchHash (ShortBranchHash)
 import qualified Unison.Codebase.ShortBranchHash as SBH
 import Unison.Prelude
+import Unison.Share.Types
 
 data ReadRepo
   = ReadRepoGit ReadGitRepo
-  | ReadRepoShare ShareRepo
-  deriving (Eq, Show)
+  | ReadRepoShare ShareCodeserver
+  deriving stock (Eq, Ord, Show)
+
+data ShareCodeserver
+  = DefaultCodeserver
+  | CustomCodeserver CodeserverURI
+  deriving stock (Eq, Ord, Show)
+
+-- |
+-- >>> :set -XOverloadedLists
+-- >>> import Data.Maybe (fromJust)
+-- >>> import Network.URI
+-- >>> displayShareCodeserver DefaultCodeserver "share" ["base", "List"]
+-- "share.base.List"
+-- >>> displayShareCodeserver DefaultCodeserver "share" []
+-- "share"
+-- >>> displayShareCodeserver (CustomCodeserver . fromJust $ parseURI "https://share-next.unison-lang.org/api" >>= codeserverFromURI ) "share" []
+-- "https://share-next.unison-lang.org:443/api:.share"
```

Comment:
I kept reading but I forgot to notice where they get printed. Though since I don't think we have parsers for these yet (before this PR or in it), I guess it doesn't matter yet?

Reply:
They're used in error messages, but no we don't have parsers for these as arguments yet, so they won't come up for users unless they pass the `UNISON_SHARE_HOST` env var